analyses/cell-type-ewings/template_notebooks/04-infercnv.Rmd

```diff
@@ -0,0 +1,455 @@
+```
+
+```{r, fig.height=10, fig.width=7}
+# Set heatmap padding option
+heatmap_padding <- 0.2
+ComplexHeatmap::ht_opt(TITLE_PADDING = grid::unit(heatmap_padding, "in"))
+# list of heatmaps looking at SingleR/ CellAssign vs tumor/normal 
+heatmap <- jaccard_matrices |>
+  purrr::imap(
+    \(celltype_mat, celltype_method) {
+      ComplexHeatmap::Heatmap(
+        t(celltype_mat), # transpose because matrix rows are in common & we want a vertical arrangement
+        col = circlize::colorRamp2(c(0, 1), colors = c("white", "darkslateblue")),
+        border = TRUE,
+        ## Row parameters
+        cluster_rows = TRUE,
+        row_title = celltype_method,
+        row_title_gp = grid::gpar(fontsize = 12),
+        row_title_side = "left",
+        row_names_side = "left",
+        row_dend_side = "right",
+        row_names_gp = grid::gpar(fontsize = 10),
+        ## Column parameters
+        cluster_columns = FALSE,
+        column_title = "",
+        column_title_gp = grid::gpar(fontsize = 12),
+        column_names_side = "bottom",
+        column_names_gp = grid::gpar(fontsize = 10),
+        column_names_rot = 90,
+        ## Legend parameters
+        heatmap_legend_param = list(
+          title = "Jaccard index",
+          direction = "vertical",
+          legend_width = unit(1.5, "in")
+        ),
+        show_heatmap_legend = celltype_method == "SingleR",
+      )
+    }) |>
+  # concatenate vertically into HeatmapList object
+  purrr::reduce(ComplexHeatmap::`%v%`) |>
+  ComplexHeatmap::draw(
+    heatmap_legend_side = "right",
+    # add a margin to the heatmap so labels don't get cut off
+    padding = unit(c(2, 20, 2, 2), "mm")
+  )
+```
+
+## Check for known CNVs in Ewing's 
+
+There are a few known copy number variations in Ewing's sarcoma: 
+
+- Gain of Chr8
+- Gain of Chr12
+- Gain of Chr1p
+- Loss of Chr16q
+
+Although these are the most frequent, there are patients who do not have any of these alterations and patients that only have some of these alterations. 
+See [Tirode et al.,](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4264969/) and [Crompton et al.](https://doi.org/10.1158/2159-8290.CD-13-1037). 
+
+Below we will plot whether or not the gain or loss is present in each cell according to `InferCNV`. 
+
+```{r}
+# umaps showing where cells are with gains/losses 
+ggplot(cnv_df, aes(x= UMAP1, y = UMAP2, color = has_dupli_chr8)) +
+  geom_point(alpha = 0.5, size = 0.5) +
+  labs(title = "Chr8 gain") +
+  theme_bw()
+
+ggplot(cnv_df, aes(x= UMAP1, y = UMAP2, color = has_dupli_chr12)) +
+  geom_point(alpha = 0.5, size = 0.5) +
+  labs(title = "Chr12 gain") +
+  theme_bw()
+
+ggplot(cnv_df, aes(x= UMAP1, y = UMAP2, color = has_dupli_chr1)) +
+  geom_point(alpha = 0.5, size = 0.5) +
+  labs(title = "Chr1 gain") +
+  theme_bw()
+
+ggplot(cnv_df, aes(x= UMAP1, y = UMAP2, color = has_loss_chr16)) +
+  geom_point(alpha = 0.5, size = 0.5) +
+  labs(title = "Chr16 loss") +
+  theme_bw()
+```
+
```

Comment:
As these plots are kind of big, I think I might either use `patchwork` to join them into a grid or maybe even just remove the `has_dupli` plots from the template altogether, and only show the proportion plots, which I think are more informative overall. (Still might use patchwork for the proportion plot... or pivot and facet)


Also, as with copykat, it is probaby worth showing this data in the context of the chromosomes that are not expected to have CNVs: A distribution for each chromosome, faceted, is probably worth doing (which would make making the UMAP plots faceted by chromosome easier as well, so you don't need patchwork)